---
title: "QC Markdown for publishing"
output:
  html_document: default
  pdf_document: default
---
#Upload of libraries and expression data
```{r loading data,message='hide',warning='hide',results='hide'}
suppressPackageStartupMessages({
  library(Seurat)
  library(dplyr)
  library(cowplot)
  library(ggplot2)
  library(pheatmap)
  library(enrichR)
  library(rafalib)
  library(ggraph)
  library(clustree)
})

#load single cell data
rna.seq<- read.table ("/Users/beghor/RNAseq/P13909-P15168-P15306-P18562_merged_gencoderpkms2_scranT.txt", sep="\t", header=TRUE)
dim(rna.seq)
rna.seq[1:5,1:5]
rownames(rna.seq) <- rna.seq[,1] 
rna.seq <- rna.seq[,-1] 
rna.seq[, 1:5]
rna.seq.copy <- rna.seq
```

#We create a seurat object from the expression matrix
```{r seurat object,message='hide',warning='hide',results='hide'}
new_data <- CreateSeuratObject(counts=rna.seq, project= "RNAseq", min.cells = 3, min.features = 200)
new_data_copy <- new_data
```

#addition of metadata created by Sebastian
```{r meta data,message='hide',warning='hide',results='hide'}
metadata<- read.csv("TCR_metadataset_March2022.csv", row.names=1)
new_data<-AddMetaData(new_data,metadata)
```

#subset the data based on flag_2A2B=1 double alpha and beta
```{r doublet filter,message='hide',warning='hide',results='hide'}
chain_filter<-subset(new_data,subset=AB_productive < 4)
dim(new_data) #3840
dim(chain_filter) #3840-3801=39
```

#Removal of ribosomal genes
```{r ribosomal genes, message='hide',warning='hide',fig.height=6,fig.width=10}
chain_filter<-PercentageFeatureSet(chain_filter, "^MT-", col.name = "percent.mt")
chain_filter<-PercentageFeatureSet(chain_filter, "^RP[SL]", col.name = "percent_ribo")
ribo_genes<-as.data.frame(grep(pattern = "^RP", x = rownames(x = chain_filter@assays$RNA), value = TRUE))

chain_filter_noribo<- chain_filter[ ! grepl("^RP", rownames(chain_filter)), ]
dim(chain_filter_noribo)

VlnPlot(chain_filter_noribo, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent_ribo","CD3G", "CD3D", "CD3E"), ncol = 3) 
```

#Filtering of cells with low quality and high mito gene expression
```{r filtering of cells, message='hide',warning='hide',fig.height=6,fig.width=10}
Tcellfilt.2<- subset(chain_filter_noribo, subset = nFeature_RNA > 1000 & nFeature_RNA < 8000 & percent.mt < 35 &
                     nCount_RNA < 1500000 & nCount_RNA > 600000 & (CD3E > 5 | CD3D > 5  | CD3G > 5))

VlnPlot(Tcellfilt.2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "percent_ribo","CD3G", "CD3D", "CD3E"), ncol = 3) 
finalfilt_noribo<-subset(Tcellfilt.2, subset = CD14 < 10 & CD19 < 10  & CD22 < 10 & CD300E< 5)

#its normal that cells numbers are different when ribosomal geens are removed and filtering is done. There might be cells just have ribosomal gene expression now they are removed by the filters
dim(finalfilt_noribo)

VlnPlot(finalfilt_noribo, features = c("nFeature_RNA", "nCount_RNA", "percent.mt", "CD3G", "CD3D", "CD3E","CD14","CD19","CD22"), ncol = 3) 
```

#Filtering of highly expressed gene MT-RNR2 and TCR genes to remove bias
```{r filtering of genes, message='hide',warning='hide',fig.height=6,fig.width=10}
#we find the highest expressed gene in our data, usually a mitochondrial gene
par(mar=c(4,8,2,1))
C <- finalfilt_noribo@assays$RNA@counts
C <-Matrix::t( Matrix::t(C) / Matrix::colSums(C) ) * 100
most_expressed <- order(apply(C,1,median),decreasing = T)[20:1]
transpose<-as.matrix(C[most_expressed, ])
transpose<-t(transpose)
boxplot(transpose,cex=.1, las=1, xlab="% total count per cell",col=scales::hue_pal()(20)[20:1],horizontal=TRUE)
#result: MT-RNR2

#Filter MT-RNR2 (a mitochondrial gene) its commonly removed.
finalfilt_noribo <- finalfilt_noribo[ ! grepl("MT-RNR2", rownames(finalfilt_noribo)), ]

#we remove TCR genes, dataframe generated by Sebastian
tcr_genes<-read.table("tcr_genes_only.csv", quote="\"", comment.char="")
dim(tcr_genes)
tcr_genes<-unlist(tcr_genes)
finalfilt_noribo<- finalfilt_noribo[!rownames(finalfilt_noribo)%in% tcr_genes,]
dim(finalfilt_noribo)
```

#Removal of data from empty wells from the plate
```{r filtering empty wells, message='hide',warning='hide',fig.height=6,fig.width=10}
EmptywellsP23<-subset(finalfilt_noribo,subset=well=="P23")
EmptywellsP24<-subset(finalfilt_noribo,subset=well=="P24")
Emptywells<-WhichCells(EmptywellsP23)
Emptywells1<-WhichCells(EmptywellsP24)
Emptywells2<-c(Emptywells,Emptywells1)

finalfilt_noribo2<-subset(finalfilt_noribo,cells = Emptywells2,invert=TRUE)

#saveRDS(finalfilt_noribo2,("finalfilt_noribo_may2022.rds"))
```

